pkgs/development/compilers/julia/generic.nix

```
@@ -50,27 +59,39 @@ stdenv.mkDerivation rec {
 
   makeFlags = [
     "prefix=$(out)"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "USE_BINARYBUILDER=1" # TODO: figure out how to build deps on darwin
+  ] ++ lib.optionals (!stdenv.isDarwin) [
     "USE_BINARYBUILDER=0"
   ] ++ lib.optionals stdenv.isx86_64 [
-    # https://github.com/JuliaCI/julia-buildbot/blob/master/master/inventory.py
-    "JULIA_CPU_TARGET=generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1)"
+    # https://github.com/JuliaCI/julia-buildkite/blob/main/utilities/build_envs.sh
+    "JULIA_CPU_TARGET=generic;sandybridge,-xsaveopt,clone_all;haswell,-rdrnd,base(1);x64-64-v4,-rdrnd,base(1)"
   ] ++ lib.optionals stdenv.isAarch64 [
-    "JULIA_CPU_TARGET=generic;cortex-a57;thunderx2t99;armv8.2-a,crypto,fullfp16,lse,rdm"
+    "JULIA_CPU_TARGET=generic;cortex-a57;thunderx2t99;carmel,clone_all;apple-m1,base(3);neoverse-512tvb,base(3)"
   ];
 
+  NIX_CFLAGS_COMPILE = [] ++ lib.optionals stdenv.isDarwin [
+    "-Wno-error=implicit-function-declaration"
+    "-Wno-error=elaborated-enum-base"
+  ];
+  # TODO: figure out how to re-enable these as errors?
+
   # remove forbidden reference to $TMPDIR
   preFixup = ''
+  '' + lib.optionalString stdenv.hostPlatform.isElf (''
     for file in libcurl.so libgmpxx.so libmpfr.so; do
       patchelf --shrink-rpath --allowed-rpath-prefixes ${builtins.storeDir} "$out/lib/julia/$file"
     done
-  '';
+  '');
 
   # tests are flaky for aarch64-linux on hydra
-  doInstallCheck = if (lib.versionOlder version "1.10") then !stdenv.hostPlatform.isAarch64 else true;
+  # some tests not working on aarch64-darwin for unrelated reasons
+  doInstallCheck = if (lib.versionOlder version "1.10") then (stdenv.isLinux && !stdenv.hostPlatform.isAarch64) else stdenv.isLinux;
```

```suggestion
  doInstallCheck = stdenv.hostPlatform.isLinux && (lib.versionAtLeast version "1.10" || !stdenv.hostPlatform.isAarch64);
```

Let's simplify the logic a bit.